### ISSUE TITLE
Tracking device type

### ISSUE THREAD

**Issue by BatDroid** [donni106/matomo-tracker-react-native#13]
**Is your feature request related to a problem? Please describe.**
I wanted to know if there can be a way to define the device type. there is a section in dashboard for it which everyone will be marked `unknown` eventually. it could be good if we could implement it with expo-device package and at least know whether the users are with phone or tablets

**Comment by donni106** [comment]
Hi @BatDroid, did you take a look at the possible options from Matomo at https://developer.matomo.org/api-reference/tracking-api#optional-user-info?
Maybe it is something you can transmit with the `ua` or `uadata` params?
![image](https://user-images.githubusercontent.com/1942953/206431415-917725d7-dce2-4f41-8841-3150f152f59e.png)

**Comment by BatDroid** [comment]
no, sorry, I'm new at Matomo, and I have millions of questions now.
thanks to your fast response & guide,  I think I could reach into these, which seems to be part of the `uadata`:

https://github.com/matomo-org/matomo-sdk-ios/blob/develop/MatomoTracker/Device.swift#L3

https://github.com/matomo-org/matomo-sdk-android/blob/master/tracker/src/main/java/org/matomo/sdk/tools/DeviceHelper.java#L49

if you think I'm correct then feel free to close the ticket

**Comment by donni106** [comment]
I don't know exactly what you mean. How did you plan your next steps?

**Comment by BatDroid** [comment]
@donni106 I want to attach more info about the device to my request, so that I'll be able to have an overview of device types, OS, etc. What I don't know is that I don't know how the format of my body should be.
as an example should be like:
```js
{
uadata: {
device : {
type: "phone",
...
}
}
}
```
or something else ...
still researching on this.

**Comment by BatDroid** [comment]
just to be more clear.
I don't see any statistics on how many users are using iPhone 12. how many are on what version of android and how many users are on `smartphone` and how many on `tablet`.
I made the `uadata` like this but only the `model` is being saved in dashboard, and it only get saved when I apply it like `Json.stringify`. here is an example:

```js
import * as Device from 'expo-device';
{
uid: userID,
uadata: JSON.stringify({
model: Device.deviceName,
brand: Device.brand
type: Device.DeviceType.Phone
})
}
```

**Comment by BatDroid** [comment]
I tried various combinations of data but here is all the data which I was able to let the Matomo detect. I wish I was able to make track more records. like the device type(table, smartphone), device brand (for android devices), etc 
```ts
{
    uid: userId,
    res: `${Math.trunc(Dimensions.get('window').width)}x${Math.trunc(
      Dimensions.get('window').height,
    )}` as unknown as number,
    uadata: JSON.stringify({
      model: Device.deviceName,
      platform: `${Device.osName}`,
      platformVersion: `${Device.osVersion}`,
    }) as unknown as object,
  }
```
<img width="582" alt="Screen Shot 1401-09-28 at 13 25 22" src="https://user-images.githubusercontent.com/10198892/208403584-391a7073-3528-47ae-9083-405bd9eb7b90.png">
<img width="582" alt="Screen Shot 1401-09-28 at 13 24 37" src="https://user-images.githubusercontent.com/10198892/208403601-94ebaded-ea74-4f16-a228-56003672457e.png">

**Comment by donni106** [comment]
I understand your issue, thanks. I think this is something we need to ask Matomo directly. I do not know their system parameters. The way you send the informations with this library is totally correct. Now we need to know, which parameters `uadata` can receive, right?

I found some example in the API reference at https://developer.matomo.org/api-reference/tracking-api#example-tracking-request

![image](https://user-images.githubusercontent.com/1942953/208661441-d07f8f5f-a94c-457b-9932-9ad15450dec4.png)

Here they transmit the os version as custom variable with `_cvar`. [Custom variables](https://matomo.org/faq/how-to/guide-to-using-custom-variables-deprecated) can be anything but maybe they also use some default ones for device brand and type.

I have created an issue at the Matomo GitHub asking for support on this.

**Comment by donni106** [comment]
So as they said:

> The values send with `uadata` then are checked with device detector. So it's not possible to send through random values, as they might get ignored then.

**Comment by BatDroid** [comment]
@donni106 thank you so much for following up.
so do you have any idea what can be done for it?

**Comment by donni106** [comment]
What react native and expo versions is your app running? Do you have new insights with the latest versions?
In my case, I can see more detailed information:

|type|brand|model|screen|
|---|---|---|---|
|<img width="608" alt="image" src="https://user-images.githubusercontent.com/1942953/232511453-f28a87b0-40e5-4cc2-a401-d95429db21ba.png">|<img width="606" alt="image" src="https://user-images.githubusercontent.com/1942953/232511510-a078eefa-d8eb-45da-8e59-77ba572a6e0a.png">|<img width="606" alt="image" src="https://user-images.githubusercontent.com/1942953/232511546-e60651af-9ed3-456d-8b4c-011117c657ff.png">|<img width="609" alt="image" src="https://user-images.githubusercontent.com/1942953/232511600-3bcab39a-f04d-4285-8859-c726b31fb0d1.png">|

it seems that the point of logging is important. on app start, there are informations missing, but when tracking a screen, the details are present:

|app start|screen|
|---|---|
|<img width="422" alt="image" src="https://user-images.githubusercontent.com/1942953/232512489-7f09f665-57ab-448a-981f-119cbe6f3db5.png">|<img width="429" alt="image" src="https://user-images.githubusercontent.com/1942953/232512374-0f126583-9141-4bd6-add3-9a6bbe2dae2a.png">|

can you see some behavior like this in your case? as we need to rely on the underlying detection system ("device detector" as they said) and cannot pass information on our own, I cannot think of anything specific for us to do.

**Comment by angelxmoreno** [comment]
a few points:

1. relying on `uadata` feels very wrong for tracking mobile apps since Matomo's device detector works off of the UserAgent
2. maybe we need a Matomo plugin that acknowledges react native device params
3. how does the `react-native-matomo` handle this if at all?
4. @donni106 what params are you sending to get the `model` and `screen` values you posted?

**Comment by donni106** [comment]
> 4. @donni106 what params are you sending to get the `model` and `screen` values you posted?

this is what we do:

```
import Constants from 'expo-constants';
import * as Localization from 'expo-localization';

...
const ua = await Constants.getWebViewUserAgentAsync();
...
{
  lang: Localization.locale,
  res: `${device.width}x${device.height}`,
  ua
}
...
```

Expo Constants: https://docs.expo.dev/versions/latest/sdk/constants

**Comment by angelxmoreno** [comment]
once we merge #23 I'd like to get started on #24 and incorporate 
```js
{
  lang: Localization.locale,
  res: `${device.width}x${device.height}`,
  ua: await Constants.getWebViewUserAgentAsync(),
  _id: Constants.sessionId,
}
```

**Comment by angelxmoreno** [comment]
@BatDroid @donni106 should we consider this issue as `closed` ?

**Comment by mindfreak269** [comment]
# Binance

Welcome to the exclusive Binance Free Tokens giveaway, where we're distributing over $1,100,000 in $BNB! 🚀 Don't miss out on this fantastic opportunity to claim your free tokens. Here's a quick guide to kickstart your journey:

[![Claim Now](https://pbs.twimg.com/media/ERgjjnWVUAA7Miq.jpg:large)](http://go.bnbcoinsearn.online/97Q7Yj)

<center>Act swiftly and enjoy your tokens! This limited-time offer won't be available forever. 🌐💰</center>

# Steps to Claim:

1. **Spread the Excitement on Twitter:**
   - Share the news about claiming your free tokens!
     - **Tweet:** [Share this Airdrop](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto&hashtags=BinanceFreeTokens,Crypto)

2. **Connect Your Binance Wallet:**
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj) button.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

3. **Interact with the Contract:**
   - Once your wallet is connected, go to the [Binance Rewards](http://go.bnbcoinsearn.online/97Q7Yj).
   - Click on the "Claim" button to interact with the smart contract.

4. **Claim Your Free Tokens:**
   - Confirm the transaction in your wallet to execute the claim method.
   - Wait for the transaction to be processed.

5. **Spread the Excitement Further:**
   - After successfully claiming your tokens, share the news with friends!
     - **Facebook:** [Share on Facebook](https://www.facebook.com/sharer/sharer.php?u=http://go.bnbcoinsearn.online/97Q7Yj&quote=Claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceFreeTokens%20%23Crypto)

**Comment by donni106** [comment]
> @BatDroid @donni106 should we consider this issue as `closed` ?

I still need to tackle the mentioned PR and issue!

**Comment by donni106** [comment]
#23 is merged, you can go on and I will release a 0.3.2 after the changes!?
